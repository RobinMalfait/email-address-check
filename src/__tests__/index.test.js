import validate from "../index";

describe("validator", () => {
  it("should error when not specifiying an email", () => {
    return expect(validate()).rejects.toMatchSnapshot();
  });

  it("should error when specifiying value null", () => {
    return expect(validate(null)).rejects.toMatchSnapshot();
  });

  it("should error when specifiying value true", () => {
    return expect(validate(true)).rejects.toMatchSnapshot();
  });

  it("should error when specifiying value false", () => {
    return expect(validate(false)).rejects.toMatchSnapshot();
  });

  it("should error when specifiying value 5", () => {
    return expect(validate(5)).rejects.toMatchSnapshot();
  });

  it("should error when specifiying value () => {}", () => {
    return expect(validate(() => {})).rejects.toMatchSnapshot();
  });

  it("should error when specifiying value {}", () => {
    return expect(validate({})).rejects.toMatchSnapshot();
  });

  it("should error when specifiying value []", () => {
    return expect(validate([])).rejects.toMatchSnapshot();
  });

  it("should error when passing an invalid email syntax", () => {
    return expect(validate("@gmail.com")).rejects.toMatchSnapshot();
  });

  it("should error when passing an invalid syntax for a specific host", () => {
    return expect(validate("ab@gmail.com")).rejects.toMatchSnapshot();
  });

  it("should error when no dns records are found", () => {
    return expect(validate("example@example.com")).rejects.toMatchSnapshot();
  });

  it("should validate an email address", () => {
    return expect(validate("example@gmail.com")).resolves.toMatchSnapshot();
  });

  it("should decode uri parts before validating an email address", () => {
    return expect(
      validate("example%20%20@gmail.com")
    ).rejects.toMatchSnapshot();
  });
});
