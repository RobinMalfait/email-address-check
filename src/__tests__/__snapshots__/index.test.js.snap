// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validator should decode uri parts before validating an email address 1`] = `[Error: E-mail address has not a valid signature]`;

exports[`validator should error when no dns records are found 1`] = `[Error: Hostname does not have any DNS records attached]`;

exports[`validator should error when not specifiying an email 1`] = `[Error: E-mail address not specified]`;

exports[`validator should error when passing an invalid email syntax 1`] = `[Error: E-mail address has not a valid signature]`;

exports[`validator should error when passing an invalid syntax for a specific host 1`] = `[Error: E-mail address did not match the rules for gmail.com]`;

exports[`validator should error when specifiying value () => {} 1`] = `[Error: E-mail address is not a string]`;

exports[`validator should error when specifiying value [] 1`] = `[Error: E-mail address is not a string]`;

exports[`validator should error when specifiying value {} 1`] = `[Error: E-mail address is not a string]`;

exports[`validator should error when specifiying value 5 1`] = `[Error: E-mail address is not a string]`;

exports[`validator should error when specifiying value false 1`] = `[Error: E-mail address is not a string]`;

exports[`validator should error when specifiying value null 1`] = `[Error: E-mail address is not a string]`;

exports[`validator should error when specifiying value true 1`] = `[Error: E-mail address is not a string]`;

exports[`validator should error when the email address contains malformed uri decoded parts 1`] = `[Error: E-mail address has not a valid signature]`;

exports[`validator should error when the hostname is blacklisted 1`] = `[Error: E-mail address has been blacklisted by spamhaus.org]`;

exports[`validator should validate an email address 1`] = `true`;
